engine/src/main/scala/cromwell/engine/backend/CromwellBackend.scala

```diff
@@ -15,24 +15,43 @@ import org.slf4j.LoggerFactory
   */
 object CromwellBackend {
   private val log = LoggerFactory.getLogger(getClass.getName)
-  private var _backend: Option[Backend] = None
+  private var _backend: Option[Map[String, Backend]] = None
+  private var _defaultBackend: Option[Backend] = None
 
-  def initBackend(backendType: String, actorSystem: ActorSystem): Backend = {
+  def initBackends(backendTypes: List[String], defaultBackend: String, actorSystem: ActorSystem) = {
+
+    _backend = Option((backendTypes map { backendType => (backendType, Backend.from(backendType, actorSystem)) }).toMap)
+    _defaultBackend = Option(Backend.from(defaultBackend, actorSystem))
+
+//    _backend match {
+//      case None =>
+//        _backend = Option((backendTypes map { backendType => (backendType, Backend.from(backendType, actorSystem)) }).toMap)
+//        _defaultBackend = Option(Backend.from(defaultBackend, actorSystem))
+//      case Some(x) if backendTypes.equals(x.keys.toList) => // TODO: This is only here because Main creates two instances of WorkflowManagerSystem. We should probably fix that.
+//      case Some(x) => throw new IllegalStateException(s"The set of backends has already been initialised to ${x.mkString}. Don't do that again (with ${backendTypes.mkString})!")
+//    }
+  }
+
+  def backend(backendName: String): Backend = _backend match {
+    case Some(x: Map[String, Backend]) => x(backendName)
+    case None => throw new IllegalStateException("backend() called prior to initBackends")
+  }
+
+  /**
+    * Register a new custom backend in the CromwellBackend set. This is mainly useful just for test cases which want
+    * to use a custom/mock/stub backend in a test.
+    *
+    */
+  def registerCustomBackend(backendName: String, backend: Backend) = {
```

Comment:
If this is for tests can it have package visibility so it's not part of the public API?